includes/Core/Modules/Module.php

```diff
@@ -662,6 +662,31 @@ final protected function permute_site_url( $site_url ) {
 		return $urls;
 	}
 
+	/**
+	 * Permutes hostname to cover all different variants of it (not considering the path).
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $site_url Site URL to get hostname permutations for.
+	 * @return array List of permutations.
+	 */
+	final protected function permute_hostname( $site_url ) {
```

Comment:
Sorry if my previous comment was unclear, but we don't need to add a new method for permuting hostnames, we can simply use the `permute_site_url` method we have now. Like I said, we just need to map the result into the hostname from the array of URLs.

It also doesn't matter which element is which, so we don't need to key them differently to identify www vs non-www. We just need to map them into the array of values, (formatted slightly different for AdSense) and provide them to the respective filters.

Happy to show you what I mean if you have any questions.

Reply:
I'm sorry, I don't understand what the first part of your comment means.

I understand the second part.